x/metadata/types/msg.go

```diff
@@ -680,6 +560,52 @@ func (msg MsgDeleteRecordSpecificationRequest) ValidateBasic() error {
 	return nil
 }
 
+// ------------------  MsgP8EMemorializeContractRequest  ------------------
+
+// NewMsgP8EMemorializeContractRequest creates a new msg instance
+func NewMsgP8EMemorializeContractRequest() *MsgP8EMemorializeContractRequest {
+	return &MsgP8EMemorializeContractRequest{}
+}
+
+func (msg MsgP8EMemorializeContractRequest) String() string {
+	out, _ := yaml.Marshal(msg)
+	return string(out)
+}
+
+// Route returns the module route
+func (msg MsgP8EMemorializeContractRequest) Route() string {
+	return ModuleName
+}
+
+// Type returns the type name for this msg
+func (msg MsgP8EMemorializeContractRequest) Type() string {
+	return TypeMsgP8eMemorializeContractRequest
+}
+
+// GetSigners returns the address(es) that must sign over msg.GetSignBytes()
+func (msg MsgP8EMemorializeContractRequest) GetSigners() []sdk.AccAddress {
+	retval := []sdk.AccAddress{}
+	if msg.Signatures != nil {
```

Comment:
So this flow won't actually work for P8e ... the do not sign the Msg with the embedded signature accounts ... they use a "notary" account which is the only signer.  This process is appropriate for extracting the actual signers for contract processing (which I noticed is being used in the Msg handler)